src/lib/Components/FilterModal.tsx

```diff
@@ -50,42 +36,172 @@ export class FilterModal extends React.Component<ModalProps, State> {
   }
 
   render() {
+    const { isFilterArtworksModalVisible } = this.props
+    const { isComponentMounted } = this.state
+
     return (
-      <RNModal animationType="fade" transparent={true} visible={this.props.visible}>
-        <TouchableWithoutFeedback>
-          <ModalBackgroundView>
+      <>
+        {isFilterArtworksModalVisible && (
+          <RNModal animationType="fade" transparent={true} visible={isFilterArtworksModalVisible}>
             <TouchableWithoutFeedback onPress={null}>
-              <>
-                <Flex onTouchStart={() => this.closeModal()} style={{ flexGrow: 1 }} />
-                <ModalInnerView visible={this.state.isComponentMounted}>
-                  <Flex flexDirection="row" justifyContent="space-between">
-                    <Flex alignItems="flex-end" mt={0.5} mb={2}>
-                      <Box onTouchStart={() => this.closeModal()}>
-                        <CloseIcon fill="black100" />
-                      </Box>
-                    </Flex>
-                    <Sans weight="medium" size="4">
-                      Filter
-                    </Sans>
-                    <Sans size="4">Clear all</Sans>
-                  </Flex>
-                  {this.props.children}
-                  <Button
-                    onPress={() => {
-                      this.closeModal()
+              <ModalBackgroundView>
+                <Flex onTouchStart={this.props.closeModal} style={{ flexGrow: 1 }} />
+                <ModalInnerView visible={isComponentMounted}>
+                  <NavigatorIOS
+                    navigationBarHidden={true}
+                    initialRoute={{
+                      component: FilterOptions,
+                      passProps: { closeModal: this.props.closeModal },
                     }}
-                    block
-                    width={100}
-                    variant="secondaryOutline"
-                  >
-                    Ok
-                  </Button>
+                    style={{ flex: 1 }}
+                  />
+                  <Box p={2}>
+                    <Button onPress={() => this.closeModal()} block width={100} variant="secondaryOutline">
+                      Apply
+                    </Button>
+                  </Box>
                 </ModalInnerView>
-              </>
+              </ModalBackgroundView>
             </TouchableWithoutFeedback>
-          </ModalBackgroundView>
-        </TouchableWithoutFeedback>
-      </RNModal>
+          </RNModal>
+        )}
+      </>
     )
   }
 }
+
+interface FilterOptionsState {
+  filterOptions: Array<{ type: string; data: any }>
```

Comment:
looks like `data` is not used anywhere?

Reply:
Not yet! It will be in the next iteration.